### DISCUSSION TITLE
Question: why not depot_tools?

### DISCUSSION THREAD

**Discussion by Skwoogey** [Eloston/ungoogled-chromium#2175] [Ideas]
A bit of yet another question about incremental builds, but i haven't quite seen the answers to the question i have in other issues, Has been looking into why and how. For some context i have only a single windows machine and it took me about a day to build chromium from scratch.

It seems as if depot_tools allow for shallow clone and shallow fetch. I have tried shallow clone and shallow fetch via git with base chromium repository / no submodules and shallow fetch from one tag to another is miles faster. I would assume depot tools has an option of switching by tags while remaining shallow or at the very least it shouldn't be too hard to edit it to that point. So i was wondering why so?

depot_tools seem to be glorified git wrapper anyway and we have to get source code from google this way or another. And using git seems to solve many problems: faster update to new versions while remaining shallow, incremental builds and with a bit of tinkering unchanged dependencies might be left patched with new versions of chromium.

My general idea of how updates would work is as follows:

1. something akin to git reset --hard to restore chromium and submodules state
2. shallow fetch to a new tag with chromium and submodules
3. apply patches and stuff again
4. build

Theoretically should be only a few hundred files to rebuild after.

So why not? is it a tinfoil hat issue to minimize talking to the google services or does it undermine the ungoogled-ness of the final chromium binary somehow?

To clarify, for me it is not as much of trust issue in the binaries provided by community, but more of a curiosity and fun adventure. I understand  that i might be blabbering nonsense here, so if i am wrong i'd gladly be educated on the matter.

**Comment by PF4Public** [comment]
> So why not? is it a tinfoil hat issue to minimize talking to the google services or does it undermine the ungoogled-ness of the final chromium binary somehow?

Those are good questions! My guess would be that neither someone did actually a side-by-side comparision of the two ways of obtaining sources, nor did anyone investigate how does this approach affect maintenance of the patches and pruning list. There is already a pretty solid workflow in place, why fix what is not [yet] broken…

>To clarify, for me it is not as much of trust issue in the binaries provided by community, but more of a curiosity and fun adventure.

It would be of immense value to the community if you could take the adventure and repot us back all your findings :) 

I can see a potential benefit of your approach in that there would be no need to wait for a tarball to start preparing the next release. There might be also a downside to it as it might be lacking PGO profiles.
